test/build-plugin.test.js

```diff
@@ -73,6 +71,30 @@ test('plugin', (t) => {
     )
   })
 
+  t.test('no namespace - secrets array', async (t) => {
+    t.plan(2)
+
+    const decorate = sinon.spy()
+
+    await plugin(
+      { decorate },
+      {
+        secrets: ['secret1-name', 'secret2-name']
+      }
+    )
+
+    t.ok(decorate.called, 'decorates fastify')
+    t.ok(
+      decorate.calledWith('secrets', {
+        0: 'content for secret1-name',
```

Comment:
I would not expect these numbers properties, don't you think they are a surplus?